src/utilities/util_dump.c

```diff
@@ -146,12 +165,34 @@ util_dump(WT_SESSION *session, int argc, char *argv[])
         ret = util_err(session, errno, NULL);
 
     __wt_scr_free(session_impl, &tmp);
+    __wt_scr_free(session_impl, &tmp2);
+    __wt_scr_free(session_impl, &timestamp_buf);
     free(uri);
     free(simpleuri);
 
     return (ret);
 }
 
+/*
+ * time_pair_to_timestamp --
+ *     Convert a timestamp output format to timestamp representation.
+ */
+static int
+time_pair_to_timestamp(WT_SESSION_IMPL *session_impl, char *ts_string, WT_ITEM *buf)
+{
+    uint64_t timestamp;
+    uint32_t first, second;
+
+    if (ts_string[0] == '(') {
+        sscanf(ts_string, "(%u,%u)", &first, &second);
+        timestamp = ((unsigned long long)first << 32) | second;
+    } else
+        timestamp = strtoul(ts_string, NULL, 10);
+
+    WT_RET(__wt_buf_catfmt(session_impl, buf, "%lx", timestamp));
+    return 0;
+}
```

Comment:
You never want to use `unsigned long long` or `%lx`, or `%u` when working with timestamps or, well, any known type size because they're undefined in size, that is, for example, a conforming implementation could have a `long` that's 4 bytes.

I'm not sure, shouldn't the `uint64_t` be a `wt_timestamp_t`? Do we export that type? If not, yeah, I guess `uint64_t` is what you've got.

Anyway, instead of `%u`, I'd use `SCNx32`, instead of `unsigned long long`, `uint64_t`, and instead of `strtoul`, `__wt_strtouq` (we have our own version in `os/os_strtouq.c` to handle the conversion issues).

We have to check the `sscanf` return for error.

No parenthesis around the `return`.

Reply:
You can find an example in `test/csuite/timestamp/abort/main.c:thread_ckpt_run` where it both does the `sscanf` and a `printf` of a timestamp.